src/libecalc/dto/graph.py

```diff
@@ -27,17 +27,38 @@ def get_successors(self, component_id: str, recursively=False) -> List[str]:
         else:
             return self.graph.successors(component_id)
 
-    def get_predecessor(self, component_id) -> str:
+    def get_predecessor(self, component_id: str) -> str:
         predecessors = list(self.graph.predecessors(component_id))
         if len(predecessors) > 1:
             raise ValueError("Component with several parents encountered.")
         return predecessors[0]
 
+    def get_parent_installation_id(self, component_id: str) -> str:
```

Comment:
recursive, find first parent that is an installation. Not completly done. Might be better ways. Perhaps make more general.